### PR TITLE
fix(Core/Arena): allow crossfaction arena teams

### PR DESCRIPTION
**Pull request by r-o-b-o-t-o**
## Changes Proposed:
- Allow requesting an arena team charter signature to a player of the opposite faction if `CONFIG_ALLOW_TWO_SIDE_INTERACTION_GUILD` is enabled
- Fixes the issue where players of the opposite faction cannot cast beneficial spells on teammates in crossfaction arena matches

## Issues Addressed:

- Fixes https://github.com/azerothcore/mod-cfbg/issues/101
- Fixes https://github.com/azerothcore/azerothcore-wotlk/issues/16821
- Fixes https://github.com/azerothcore/azerothcore-wotlk/issues/16822
- Fixes https://github.com/chromiecraft/chromiecraft/issues/5900
- Fixes https://github.com/chromiecraft/chromiecraft/issues/5971

## Tests Performed:

- Compiled and tested on Windows 11

## How to Test the Changes:

- Join skirmish with two crossfaction groups, try to buff/heal your teammates.
- Create an arena team charter. Invite a player of the other faction in a group, request a charter signature.

## Known Issues and TODO List:

- The changes in `Arena.cpp` feel very hacky, feel free to suggest better alternatives

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by heyitsbench** [comment]
Should it not use a new config option of its own rather than the guild one?

**Comment by heyitsbench** [comment]
Also might want to include the CC issues in the addressed issues section as well.

**Comment by r-o-b-o-t-o** [comment]
> Should it not use a new config option of its own rather than the guild one?

I used the guild one because it is what was already used when handling charter signups: https://github.com/azerothcore/azerothcore-wotlk/blob/a5e84f610a0e5fb83c58256676f98cc1b387e3d0/src/server/game/Handlers/PetitionsHandler.cpp#L418-L421
I agree, that's not ideal. I might create a new config option tomorrow.

File: src/server/game/Battlegrounds/Arena.cpp [hunk 1]
```diff
@@ -93,6 +94,47 @@ void Arena::AddPlayer(Player* player)
...
+            group->Disband();
+
+            group = new Group();
+            SetBgRaid(player->GetBgTeamId(), group);
+            group->Create(player);
+            sGroupMgr->AddGroup(group);
+            for (Player* member : members)
+            {
+                group->AddMember(member);
+            }
```
Comment (sassouiabd) [inline src/server/game/Battlegrounds/Arena.cpp:135]: Smart idea ;)

**Review by Nefertumm — APPROVED** [review]

**Pull request merged**